Lib/test/test_tempfile.py

```diff
@@ -1826,17 +1865,28 @@ def test_modes(self):
 
     @unittest.skipUnless(hasattr(os, 'chflags'), 'requires os.lchflags')
     def test_flags(self):
-        flags = stat.UF_IMMUTABLE | stat.UF_NOUNLINK
-        d = self.do_create(recurse=3, dirs=2, files=2)
-        with d:
-            # Change files and directories flags recursively.
-            for root, dirs, files in os.walk(d.name, topdown=False):
+        def chflags_recursively(name, flags):
+            for root, dirs, files in os.walk(name, topdown=False):
                 for name in files:
                     os.chflags(os.path.join(root, name), flags)
                 os.chflags(root, flags)
-            d.cleanup()
-        self.assertFalse(os.path.exists(d.name))
 
+        d = self.do_create(recurse=3, dirs=2, files=2)
+        try:
+            with d:
+                chflags_recursively(d.name, stat.UF_IMMUTABLE | stat.UF_NOUNLINK)
+                d.cleanup()
+
+            self.assertFalse(os.path.exists(d.name))
+
+        finally:
+            # Even if test fails, make a best-effort attempt at not leaving
```

Comment:
I'm not convinced by this change. Tests that pass should not leave behind cruft, but it's common for tests that fail to do so. And in general I think it's better if failing tests leave behind _unmodified_ cruft, because modifications may confuse debugging of the test failure.

Since IMO this change is not obviously an improvement, it might be better for it to be in a separate PR/issue where it can be discussed on its own merits, instead of tacked onto an unrelated PR?

Reply:
Removed.